armada/armada.py

```diff
@@ -59,37 +61,57 @@ def sync(self):
             protoc_chart = chartbuilder.get_helm_chart()
 
             # determine install or upgrade by examining known releases
+            LOG.debug("RELEASE: %s", chart.release_name)
+
             if chart.release_name in [x[0] for x in known_releases]:
 
                 # indicate to the end user what path we are taking
                 LOG.info("Upgrading release %s", chart.release_name)
-
                 # extract the installed chart and installed values from the
                 # latest release so we can compare to the intended state
                 installed_chart, installed_values = self.find_release_chart(
                     known_releases, chart.release_name)
 
+                if not self.args.disable_update_pre:
+                    try:
```

Comment:
Could shorten this

`pre_actions = getattr(chart.upgrade, 'pre', {})`

By intercepting the excepton, but still referencing pre_actions below you'd likely encounter an issue if `pre` was actually absent.